pype_schema/parse_json.py

```diff
@@ -692,23 +784,28 @@ def parse_virtual_tag(tag_id, tag_info, obj):
             parent object that contains all constituent tags,
             which is used to gather the tag list for combining data correctly
 
+        parent_network : None, Network
+            Optional network object the tag is a part of
+            If `None` will assume `obj` is the parent network and all tags are in `obj.tags`
+
         Returns
         -------
         VirtualTag
             a Python object with the given ID and the values from `tag_info`
         """
         tag_list = []
+        parent_network = obj if parent_network is None else parent_network
         for subtag_id in tag_info["tags"]:
-            if isinstance(obj, connection.Connection):
-                subtag = obj.get_tag(subtag_id)
-            else:
-                subtag = obj.get_tag(subtag_id, recurse=True)
+            subtag = parent_network.get_tag(subtag_id, recurse=True)
             if subtag is None:
                 raise ValueError(
                     "Invalid Tag id {} in VirtualTag {}".format(subtag_id, tag_id)
                 )
             tag_list.append(subtag)
-
+        try:
+            pint_unit = utils.parse_units(tag_info["units"])
+        except:
```

Comment:
Why do we have to catch the error here? I think there's probably a better way to deal with this upstream since the errors thrown by our own function `parse_units` should be intelligent (hopefully)